caffe2/operators/pack_segments.cu

```diff
@@ -186,15 +186,31 @@ bool PackSegmentsOp<CUDAContext>::DoRunWithType2() {
   // Find the length of the longest sequence.
   dev_max_length_.Resize(1);
   host_max_length_.Resize(1);
-  const T max_length = num_seq > 0 ? array_max<T>(
-                                         lengths_ptr,
-                                         num_seq,
-                                         dev_buffer_,
-                                         dev_max_length_,
-                                         host_max_length_,
-                                         context_)
-                                   : 0;
 
+  T temp;
```

Comment:
why here needs to be changed to call array_max  in both branches of an if-else? can't it be changed like in cpu implementation?
```
if (max_length_ != -1) {
    // Final dim must be greater than the max_length
    CAFFE_ENFORCE_GE(max_length_, max_length, "...");
    max_length = max_length_;
}
```

Reply:
I guess I wanted to deal with the const that time. I will change it